benchmarks/NodeRegression/utils.py

```diff
@@ -0,0 +1,289 @@
+        self.best_score = None
+        self.early_stop = False
+        self.early_stop_flag = early_stop
+        self.dir_name = "checkpoints/"
+        if ~os.path.isdir(self.dir_name):
+            os.makedirs(self.dir_name, exist_ok=True)
+        ckpt_name = ckpt_name.replace("/", "_")
+        ckpt_name = os.path.splitext(ckpt_name)[0]
+        self.ckpt_dir = self.dir_name + ckpt_name + "_checkpoint.pt"
+
+    def step(self, loss, model):
+        """Step early stopping."""
+        score = loss
+        if self.best_score is None:
+            self.best_score = score
+            self.save_checkpoint(model)
+        elif score > self.best_score:
+            if self.early_stop_flag:
+                self.counter += 1
+                print(f"EarlyStopping counter: {self.counter}\
+                        out of {self.patience}")
+                if self.counter >= self.patience:
+                    self.early_stop = True
+        else:
+            self.best_score = score
+            self.save_checkpoint(model)
+            self.counter = 0
+        return self.early_stop
+
+    def save_checkpoint(self, model):
+        """Save model when validation loss decrease."""
+        torch.save(model.state_dict(), self.ckpt_dir)
+
+
+def makedirs_rm_exist(dir_name):
+    """Make a directory, remove any existing data."""
+    if os.path.isdir(dir_name):
+        shutil.rmtree(dir_name)
+    os.makedirs(dir_name, exist_ok=True)
+
+
+def set_seed(seed):
+    """Set random seed."""
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed(seed)
+
+
+def eval_rocauc(y_pred, y_true):
+    """Evalution function for ROC."""
+    rocauc_list = []
+    y_true = y_true.detach().cpu().numpy()
+    if len(y_true.shape) > 1:
+        if y_true.shape[1] == 1:
+            # use the predicted class for single-class classification
+            y_pred = F.softmax(y_pred, dim=-1)[:, 1].unsqueeze(1).cpu().numpy()
+        else:
+            y_pred = y_pred.detach().cpu().numpy()
+
+        for i in range(y_true.shape[1]):
+            # AUC is only defined when there is at least one positive data.
+            if np.sum(y_true[:, i] == 1) > 0 and np.sum(y_true[:, i] == 0) > 0:
+                is_labeled = y_true[:, i] == y_true[:, i]
+                score = roc_auc_score(y_true[is_labeled, i],
+                                      y_pred[is_labeled, i])
+
+                rocauc_list.append(score)
+    else:
+        y_pred = y_pred.detach().cpu().numpy()
+        is_labeled = ~torch.isnan(torch.tensor(y_true))
+        score = roc_auc_score(y_true[is_labeled], y_pred[is_labeled, 1])
+        rocauc_list.append(score)
+
+    if len(rocauc_list) == 0:
+        raise RuntimeError(
+            "No positively labeled data available. Cannot compute ROC-AUC.")
+
+    return sum(rocauc_list)/len(rocauc_list)
+
+
+def check_binary_classification(dataset):

```

Comment:
Let's do some cleaning on the redundant functions that are not needed for regression tasks. Please also change "classification" to "regression" in function names and comments.